### ISSUE TITLE
[FEATURE] separate Presentation layer code from Session & Transport layers

### ISSUE THREAD

**Issue by skewty** [matteosox/pysparkplug#13]
## What seems to be the problem here? 🐛
Looking to use the OSI layer 6 ([Presentation layer](https://www.imperva.com/learn/application-security/osi-model/)) code without the lower OSI layers coming along with it.

## Proposed solution 👩‍💻
Split code into two modules. One for Presentation layer and one for Session / Transport (MQTT) layers.

**Comment by skewty** [comment]
Basically looking to remove the legacy (not required for me) `paho-mqtt < 2` requirement.
We have an async code base that doesn't use that old version of paho-mqtt.

**Comment by matteosox** [comment]
Hi @skewty , I see you closed this as completed. What do you mean by that? I think the idea you've discussed here is a good one, but as an open-source maintainer, I'm not up to the task of such a refactor.

**Comment by skewty** [comment]
I looked at the amount of work to accomplish the task and weighed it against re-creating the parts we actually need and felt we could just create our own enums using python 3.12 StrEnum and also using pydantic models and perhaps using / contributing to pure protobuf instead would be a better long term strategy. That said we don't even need the protobuf part right now so we could additionally cut that dependency.  So I felt our use case is perhaps best solved by us.  I respect OSS developer time .. so I closed the ticket.
